consensus/types/src/attestation.rs

```diff
@@ -625,13 +625,17 @@ impl SingleAttestation {
     pub fn to_indexed<E: EthSpec>(&self, fork_name: ForkName) -> IndexedAttestation<E> {
         if fork_name.electra_enabled() {
             IndexedAttestation::Electra(IndexedAttestationElectra {
-                attesting_indices: vec![self.attester_index].into(),
+                attesting_indices: vec![self.attester_index]
+                    .try_into()
+                    .expect("Single index should not exceed MaxValidatorsPerSlot"),
```

Comment:
This probably fails clippy, maybe we should just make this method return a `Result`. It probably isn't used in that many places

Reply:
I tried that initially, but `to_indexed` gets used in attestation verification and then in some networking code and it ended up being really ugly with multiple new error variants needing to be defined just for this one error which can never occur anyway. The `expect` felt like the lesser of two evils.